receiver/hostmetricsreceiver/factory.go

```diff
@@ -89,6 +98,16 @@ func createMetricsReceiver(
 	)
 }
 
+func createLogsReceiver(
+	_ context.Context, set receiver.Settings, cfg component.Config, consumer consumer.Logs,
+) (receiver.Logs, error) {
+	oCfg := cfg.(*Config)
+
+	r := newHostMetricsReceiver(oCfg, &set)
+	r.nextLogs = consumer
+	return r, nil
```

Comment:
No need to follow the pattern of a shared component
```suggestion
	return &hostMetricsReceiver{
		cfg:      cfg.(*Config),
		nextLogs: consumer,
		settings: set,
	}, nil
```